lib/internal/modules/esm/loader.js

```diff
@@ -497,18 +499,22 @@ function getHooksProxy() {
  * Register a single loader programmatically.
  * @param {string} specifier
  * @param {string} [parentURL]
- * @returns {void}
+ * @param {any} [data] Arbitrary data passed to loader's `initialize` hook
+ * @param {any[]} transferList Objects in `data` that are changing ownership
```

Comment:
If `data` is optional, `transferList must also be optional:

```suggestion
 * @param {any[]} [transferList] Objects in `data` that are changing ownership
```